### PR TITLE
#166575803 Fix major problem in the Navbar component

### PR DESCRIPTION
**Pull request by coolbeatz71**
#### What does this PR do?
Fix major problem in the Navbar component
 
#### Description of Task to be completed?
- incorporate the ProfileDropdown component
- fix transparent background in the navbar
- make the navbar component responsive
- link the component with the SideBar component with hamburger-menu
- fix test for the component and its children
- make sure the component is connected to redux
#### How should this be manually tested?
- npm test
#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#166575803](https://www.pivotaltracker.com/story/show/166575803)

#### Screenshots (if appropriate)

N/A

#### Questions:

N/A

### CODE REVIEW

File: src/Components/NavBar/NavBar.scss [hunk 1]
```diff
@@ -1,5 +1,155 @@
+@import '../../assets/sass/components';
+@import '../../assets/sass/global';
 .navbar {
+  background: $color-light;
```
Comment (gracelungu) [inline src/Components/NavBar/NavBar.scss]: @coolbeatz71 , This color appears not to be pure white color, and I don't see you providing a way of changing this dynamically.

File: src/Components/Common/SearchBox/SearchBox.scss [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
     border: 0px solid rgba(0, 0, 0, 0.1);
     background: $color-light-soft;
-    width: 300px !important;
+  }
+
+  .input-group {
+    padding: 0 !important;
   }
   .btn-outline-primary {
     color: $color-primary;
```
Comment (gracelungu) [inline src/Components/Common/SearchBox/SearchBox.scss:25]: @coolbeatz71 Could you remove the `outline` on these 3 properties: The search box, the drop-down, the search button, and make the search box a little bit wider as we previously designed.

File: src/Components/Common/Profile/ProfileDropdown.scss [hunk 3]
```diff
@@ -13,4 +16,52 @@
     padding: 0 8px;
     font-size: 0.9em;
   }
+
+  .dropdown-menu {
+    position: absolute;
+  }
+}
+
+#menu {
```
Comment (gracelungu) [inline src/Components/Common/Profile/ProfileDropdown.scss:25]: @coolbeatz71 Could you also reduce the size of the hamburger menu on mobile display?

**Review by gracelungu — CHANGES_REQUESTED** [review]

**Review by dusmel — CHANGES_REQUESTED** [review]
Can you add screenshots to have an idea of the design

File: src/Components/NavBar/NavBar.jsx [hunk 4]
```diff
@@ -125,20 +185,31 @@ export class NavBar extends Component {
...
 export const mapDispatchToProps = dispatch => ({
-  onToggleSideNav: isDrawerDisplay => dispatch(toggleSideNav(isDrawerDisplay)),
+  onToggleSideNav: () => dispatch(toggleSideNav()),
 });
 
 NavBar.propTypes = {
   onToggleSideNav: PropTypes.func.isRequired,
   navbar: PropTypes.object.isRequired,
   currentUser: PropTypes.object.isRequired,
+  profile: PropTypes.object.isRequired,
```
Comment (dusmel) [inline src/Components/NavBar/NavBar.jsx:206]: does this mean we have to add profile prop when calling the Navbar component?

File: src/Components/NavBar/NavBar.jsx [hunk 5]
```diff
@@ -84,25 +144,24 @@ export class NavBar extends Component {
...
-          <Navbar.Brand>
+        <Navbar fixed="top" className="flex-nowrap flex-row" expand="lg" variant="dark">
+          <Navbar.Brand className="d-none d-md-block">
             <Logo />
           </Navbar.Brand>
-          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
-          <Navbar.Collapse id="responsive-navbar-nav">
+          {this.renderHamburgerIcon(faBars)}
+          {this.displaySearchBox ? (
             <Nav className="mr-auto search-container">
```
Comment (dusmel) [inline src/Components/NavBar/NavBar.jsx:153]: I noticed that the select option on search was not working did you try to fix that

**Review by dusmel — CHANGES_REQUESTED** [review]

File: src/Components/NavBar/NavBar.jsx (hunk 4 above)
Comment (coolbeatz71) [reply src/Components/NavBar/NavBar.jsx:206]: Yes. @dusmel we agreed with @Karlmusingo, @giseleiradu that data from the user should come from one object called `profile` inside the initialState which looks like this:

```
const initialState = {
  currentUser: {
    isAuth: checkToken(),
    user: checkToken() || null,
    notification: {
      notificationList: [],
      notificationsCount: 0,
    },
  },
  navbar: {
    isDrawerDisplay: false,
  },
  profile: {
    user: {
      username: '@coolbeatz71',
      firstName: '',
      lastName: '',
      email: '',
      bio: undefined,
      image: undefined,
      followers: 0,
      following: 0,
      articles: 0,
      allowEmailNotification: false,
    },
  },
};
```

File: src/Components/NavBar/NavBar.jsx (hunk 5 above)
Comment (coolbeatz71) [reply src/Components/NavBar/NavBar.jsx:153]: Yes @dusmel I am fixing it now

File: src/Components/Common/Profile/ProfileDropdown.scss (hunk 3 above)
Comment (coolbeatz71) [reply src/Components/Common/Profile/ProfileDropdown.scss:25]: Ok, let me reduce it

File: src/Components/Common/SearchBox/SearchBox.scss (hunk 2 above)
Comment (coolbeatz71) [reply src/Components/Common/SearchBox/SearchBox.scss:25]: Yes @gracelungu I have removed it

File: src/Components/NavBar/NavBar.scss (hunk 1 above)
Comment (coolbeatz71) [reply src/Components/NavBar/NavBar.scss]: @gracelungu I changed it now

**Review by dusmel — APPROVED** [review]

**Review by gracelungu — APPROVED** [review]

**Pull request merged**